lab-jeff/lib/server.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const net = require('net');
+const EE  = require('events').EventEmitter;
+
+
+let serverPort = 3000;
+let pool = {};
+pool.users = {};
+
+pool.ee = new EE();
+pool.ee.on('register', function(socket){
+  var newClient = new Client(socket);
+  pool.users[newClient.id] = newClient;
+
+  console.log(newClient.nickname + ' connected to the server.');
+  socket.write('Welcome to the chat ' + newClient.nickname + '\r\n');
+  socket.pipe(process.stdout);
+});
+
+function Client(socket) {
+  this.id = socket.remotePort;
+  this.socket = socket;
+  this.nickname = 'Guest';
```

Comment:
You seem to be creating a unique id every time a new client/socket is registered, but you're assigning the same nickname, 'Guest', to every client. It's hard to tell who is sending what message when they're all nicknamed Guest.